tracing-appender/src/lib.rs

```diff
@@ -1,15 +1,80 @@
+//! Writers for logging events/spans
+//!
+//! # Overview
+//!
+//! [`tracing`][tracing] is a framework for instrumenting Rust programs to collect
+//! structured, event-based diagnostic information. This crate provides the ability
+//! for `tracing` events and spans to be recorded in a non blocking manner by using a
+//! dedicated logging thread. It also provides a [`RollingFileAppender`][file_appender]
+//! that can be used with or without the non blocking writer.
+//!
+//! [file_appender]: ./rolling/struct.RollingFileAppender.html
+//! [tracing]: https://docs.rs/tracing/
+//!
+//! # Usage
+//! This crate can be used in a few ways to record spans/events:
+//! 1. Using a `RollingFileAppender` to perform writes to a log file. This will block on writes.
+//! 2. Using *any* `std::io::Write` implementation in a non-blocking way
+//! 2. Using a combination of `NonBlocking` and `RollingFileAppender` to allow writes to a log file
```

Comment:
Maybe just make this a bulleted list:
```suggestion
//!  - Using a `RollingFileAppender` to perform writes to a log file. This will block on writes.
//!  - Using *any* `std::io::Write` implementation in a non-blocking way
//!  - Using a combination of `NonBlocking` and `RollingFileAppender` to allow writes to a log file
```
and possibly link to the appropriate types/modules here?